decentraland/unity-explorer
# fix: emote lazy loading

## What does this PR change?

Fixes #2298

We do not request all the emotes of all the remote avatars anymore. Instead we request the emotes whenever they need to be played. For the **main player only** we keep requesting all of which are described in the `profile.emotes`.
This change should greatly reduce the amount of requests we make to the content server / asset bundles, specially when we are in places on which there is a lot people around. It should reduce the memory consumption and unlock many processes that were stuck due to budgeting.

## How to test the changes?

1. Launch the explorer
2. Check that you can play all your emotes
3. Get in a group of people and start playing emotes
4. You should be able to see your emotes and the ones from the remote avatars too
5. Go to the backpack and change your emotes
6. Check that emotes play in the backpack
7. Check that all emotes are described as expected
8. Play the new equipped emotes in-world
9. Open the passport the passport of another player
10. The emotes and wearables should be listed accordingly

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
